src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -275,9 +294,8 @@ Copyright (c) .NET Foundation. All rights reserved.
           Inputs="@(_ReadyToRunCompileList)"
           Outputs="%(_ReadyToRunCompileList.OutputR2RImage)">
 
-    <!-- TODO: BUG - FIX ImplementationAssemblies to take managed implementation assemblies to use as crossgen references -->
     <RunReadyToRunCompiler CrossgenTool="@(_CrossgenTool)"
-                           ImplementationAssemblies="@(ResolvedFileToPublish)"
+                           ImplementationAssemblies="@(_ReadyToRunImplementationAssemblies)"
```

Comment:
This `_ReadyToRunImplementationAssemblies` would be the same one we pass to the ILLinker, correct?

Reply:
Not necessarily. These are all from the runtime pack. There is talk of supporting portable apps. I'm leaving the task of generating the right lists to #3109. This is purely a fix for framework-dependent + crossgen. Crossgen still requires a RID always, linker may not. I think there will be some common set of items to both, but the input to the tasks probably cannot be identical.

Reply:
I will refactor code to share the commonalities. Basically build up the common list, then add/remove as necessary for each tool or something. But let's save this for the discussion on the fix for #3109.